src/cluster/index.js

```diff
@@ -105,6 +105,7 @@ module.exports = class Cluster {
    * @returns {Promise}
    */
   async refreshMetadata() {
+    await this.connect()
```

Comment:
This is kind of an aside, but the `connect` method won't actually cycle through all brokers when trying to set up the seed broker. It will obey the user defined retry configuration, which by default means that it will retry 5 of the brokers (or cycle through the same broker multiple times if brokers < retries). I think we should change `cluster.connect` to use a retrier with retries = brokers - 1.

Reply:
This logic was moved to the `BrokerPool` and doesn't rely on the seed brokers anymore.